modules/consensus/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+use futures::channel::mpsc::SendError as FuturesSendError;
+use futures::Sink;
+use futures::SinkExt;
+use futures_util::StreamExt;
+use solana_lite_rpc_core::block_store::BlockInformation;
+use solana_sdk::epoch_info::EpochInfo;
+use solana_transaction_status::EncodedConfirmedBlock;
+use std::collections::HashMap;
+use thiserror::Error;
+use tokio::pin;
+use tokio::task::JoinHandle;
+use yellowstone_grpc_client::GeyserGrpcClient;
+use yellowstone_grpc_client::GeyserGrpcClientError;
+use yellowstone_grpc_proto::geyser::CommitmentLevel;
+use yellowstone_grpc_proto::prelude::SubscribeRequestFilterAccounts;
+use yellowstone_grpc_proto::prelude::SubscribeRequestFilterBlocks;
+use yellowstone_grpc_proto::prelude::{subscribe_update::UpdateOneof, SubscribeRequestFilterSlots};
+
+#[derive(Debug, Error)]
+pub enum HistoryError {
+    #[error("Error during during send slot '{0}'")]
+    SendSlotError(String),
+
+    #[error("IO Error during history processing '{0}'")]
+    IOError(#[from] std::io::Error),
+
+    #[error("Geyser Error during history processing '{0}'")]
+    GeyserError(#[from] GeyserGrpcClientError),
+}
+
+pub async fn start_module(
+    grpc_url: String,
+    slot_sinck: impl Sink<u64, Error = FuturesSendError> + std::marker::Send + 'static,
+    full_block_sinck: impl Sink<EncodedConfirmedBlock, Error = FuturesSendError>
+        + std::marker::Send
+        + 'static,
+    block_info_sinck: impl Sink<BlockInformation, Error = FuturesSendError>
+        + std::marker::Send
+        + 'static,
+    epoch_sinck: impl Sink<EpochInfo, Error = FuturesSendError> + std::marker::Send + 'static,
+) -> JoinHandle<Result<(), HistoryError>> {
+    let handle = tokio::spawn(async move {
+        pin!(full_block_sinck);
+        pin!(block_info_sinck);
+
+        //subscribe Geyser grpc
+        let mut slots = HashMap::new();
+        slots.insert("client".to_string(), SubscribeRequestFilterSlots {});
+
+        let mut accounts: HashMap<String, SubscribeRequestFilterAccounts> = HashMap::new();
+
+        accounts.insert(
+            "client".to_owned(),
+            SubscribeRequestFilterAccounts {
+                account: vec![],
+                owner: vec![
+                    solana_sdk::stake::program::ID.to_string(),
+                    solana_sdk::vote::program::ID.to_string(),
+                ],
+                filters: vec![],
+            },
+        );
+
+        let mut blocks = HashMap::new();
+        blocks.insert(
+            "client".to_string(),
+            SubscribeRequestFilterBlocks {
+                account_include: vec![], //vec![SYSTEM_PROGRAM.to_string()],
+                include_transactions: Some(true),
+                include_accounts: None,
+                include_entries: None,
+            },
+        );
+
+        let mut client = GeyserGrpcClient::connect(grpc_url, None::<&'static str>, None)?;
+        let mut confirmed_stream = client
```

Comment:
although coupling by technology (geyser) makes some sense here I wonder if we should isolate the different sources on a per-type-basis a much as possible - i.e. one service/submodule for slots, one for blocks, one for accounts